libs/ui/src/diagnostics/uninterruptible_power_supply.tsx

```diff
@@ -0,0 +1,35 @@
+import { H2, P } from '../typography';
+import { Button } from '../button';
+import { ButtonBar } from '../modal';
+import { Main } from '../main';
+import { Screen } from '../screen';
+
+export interface UpsSectionProps {
+  passTest: () => void;
+  failTest: () => void;
+}
+
+export function UninterruptiblePowerSupplyScreen({
+  passTest,
+  failTest,
+}: UpsSectionProps): JSX.Element {
+  return (
+    <Screen>
+      <Main flexColumn padded>
+        <H2>Uninterruptible Power Supply Test</H2>
+        <P>
+          Confirm the uninterruptible power supply is connected and fully
+          charged.
+        </P>
+        <ButtonBar style={{ marginTop: '0.5rem' }}>
+          <Button icon="Done" onPress={passTest}>
+            UPS is Fully Charged
+          </Button>
+          <Button icon="Delete" onPress={failTest}>
+            UPS is not Fully Charged
+          </Button>
```

Comment:
nit: For true title case, I think we want `UPS Is Fully Charged` and `UPS Is Not Fully Charged` (because I can never remember exact title case rules I used https://titlecaseconverter.com)